### PR TITLE
feat(filter): update to latest patternfly designs

### LINKED ISSUE #553: Update Filter to latest PatternFly designs
Currently the filter component:
1. Has the placeholder text within the dropdown, remove
2. Dropdown shows italicized current value.  Only placeholder should be italicized
3. Dropdown filter can be applied multiple times, restrict to single select/filter application
4. Dropdown doesn't remember previous selection when switching between filter fields.
5. Dropdown should revert to placeholder when it's filter is removed.

@mcarrano 

Related PR: https://github.com/patternfly/patternfly-ng/pull/449

### PR DESCRIPTION
**Pull request by dtaylor113**
**What**:
Updated the filter component to latest designs outlined [here](https://patternfly.atlassian.net/browse/OSUX-740):
1. Removed placeholder text within the dropdowns
2. Placeholder are italicized, selected values are not
3. Restricted Select/Dropdown filters to single select/filter application
4. Dropdowns remember previous selection when switching between filter fields.
5. Dropdowns revert to placeholder when it's filter is removed.
6. Fixed filter unit tests
7. Removed eslint warnings from `_compound-label.scss`

@mcarrano 

Related PR: https://github.com/patternfly/patternfly-ng/pull/449

**Link to Storybook**:
[Storybook: Filter](https://dtaylor113.github.io/patternfly-react/?selectedKind=patternfly-react%2FForms%20and%20Controls%2FFilter&selectedStory=Filter&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybooks%2Fstorybook-addon-knobs)
* Here is a screen recording:
react-filter

**Additional issues**:
Fixes #553

### CODE REVIEW

File: packages/patternfly-react/less/filter.less [hunk 1]
```diff
@@ -0,0 +1,4 @@
+.filter-selected {
+  font-style: normal;
+  color: initial;
+}
```
[the remark this answers is not in the document]
Comment (dtaylor113) [reply packages/patternfly-react/less/filter.less]: Ok, sounds good.  What test pages though?  I can't select anything here: https://rawgit.com/patternfly/patternfly/master-dist/dist/tests/filter.html#

[the remark this answers is not in the document]
Comment (jeff-phillips-18) [reply packages/patternfly-react/less/filter.less]: that page shows the filter categories with selected items under the `Category Filters`

[the remark this answers is not in the document]
Comment (dtaylor113) [reply packages/patternfly-react/less/filter.less]: Going to look at implementing these updates in PF core. Leaving this PR [WIP] until PF core changes are merged.

**Comment by dtaylor113** [comment]
Waiting on https://github.com/patternfly/patternfly/pull/1137

File: packages/patternfly-react/sass/patternfly-react/_compound-label.scss [hunk 2]
```diff
@@ -25,7 +21,7 @@
   .list-inline {
     display: inline-block;
 
-    &>li {
+    & > li {
```
Comment (jeff-phillips-18) [inline packages/patternfly-react/sass/patternfly-react/_compound-label.scss]: this file shouldn't need to be included, rebasing should resolve the issues

Comment (dtaylor113) [reply packages/patternfly-react/sass/patternfly-react/_compound-label.scss]: ? I changed this file to get rid of eslint warnings during builds.

Comment (jeff-phillips-18) [reply packages/patternfly-react/sass/patternfly-react/_compound-label.scss]: it has been fixed already

Comment (dtaylor113) [reply packages/patternfly-react/sass/patternfly-react/_compound-label.scss]: ok, rebased, it did go away -thanks

**Comment by dtaylor113** [comment]
@priley86 do you have some time to review today?  -thanks

**Pull request merged**